### PR TITLE
Surgical incisions cause actual bleeding; clamping bleeders and/or cauterizing stops that bleeding.

### PR DESCRIPTION
**Pull request by harmonyn**
## About The Pull Request

What it says in the title. For humans who are being operated on, surgical incisions will cause them to bleed. Clamping bleeders will reverse this, as will closing the incision with a cautery (either as a step or to cancel the server.)

The amount of bleeding is slightly less than other bleeding-causing things in the code, and if someone is doing surgery at a reasonable speed it shouldn't cause too much trouble, even if it isn't the immediate next step.

## Why It's Good For The Game

Because this is literally what clamping bleeders means -- that's why they're called bleeders -- and now that we have a bleeding system it seems silly not to reflect that. 

## Changelog
:cl: bandit
tweak: Surgical incisions cause actual bleeding now. Clamping bleeders and/or cauterizing the incision stops that bleeding.
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
If someone is bleeding from a prior source, will the bleeders stop it?

If you cancel the surgery using the cautery, will that stop the surgery bleeding?

**Comment by Tlaltecuhtli** [comment]
>step 1 causes bleeding
>step 2 stops  the bleeding

👌

**Comment by vpellen** [comment]
Are _all_ surgery incisions followed by clamp step? I could have sworn there were some that aren't.

**Comment by Tlaltecuhtli** [comment]
idea: bleeder clamping is no longer a surgery step, but when you hit someone with it, it temporary stops surgery induced bleeding (for 5s?) and all other steps induce extra bleeding

**Comment by bgobandit** [comment]
>If someone is bleeding from a prior source, will the bleeders stop it?

no, clamping bleeders either subtracts the added bleeding from the incision or sets bleeding to 0, whichever is higher (i.e. so there's no negative bleeding)

> If you cancel the surgery using the cautery, will that stop the surgery bleeding?

not currently but it probably should, will do

>Are all surgery incisions followed by clamp step? I could have sworn there were some that aren't.

no but they probably should now, they would IRL as well

File: code/modules/surgery/organic_steps.dm [hunk 1]
```diff
@@ -16,6 +16,14 @@
 
 	return TRUE
 
+/datum/surgery_step/incise/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if ishuman(target)
+		var/mob/living/carbon/human/H = target
+		if (!(NOBLOOD in H.dna.species.species_traits))
+			user.visible_message("Blood pools around the incision in [H]'s [parse_zone(target_zone)].", "<span class='notice'>Blood pools around your incision in [H]'s [parse_zone(target_zone)].</span>")
```
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/modules/surgery/organic_steps.dm]:
```suggestion
			H.visible_message("Blood pools around the incision in [H]'s [parse_zone(target_zone)].", "<span class='notice'>Blood pools around the incision in your [parse_zone(target_zone)].</span>")
```

**Comment by PKPenguin321** [comment]
Ideally you would be able to hold the hemostat in your offhand while performing incisions to clamp bleeders as you go

**Comment by Tlaltecuhtli** [comment]
make hemostat on hit give you status effect "clamped" which temporary stops your bleeding,during surgery bandages shoudlnt work

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Creative liberties can be made to my suggestion but it needs to be fixed.

File: code/modules/surgery/organic_steps.dm [hunk 2]
```diff
@@ -29,9 +37,11 @@
 /datum/surgery_step/clamp_bleeders/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	if(locate(/datum/surgery_step/saw) in surgery.steps)
 		target.heal_bodypart_damage(20,0)
+	if ishuman(target)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/organic_steps.dm]:
```suggestion
	if (ishuman(target))
```

sorry this was bothering me

**Pull request merged**